src/openvpn/options.c

```diff
@@ -2382,6 +2413,25 @@ options_postprocess_verify_ce(const struct options *options, const struct connec
                 msg(M_USAGE, "--auth-user-pass-optional %s", postfix);
             }
         }
+
+#ifdef ENABLE_VLAN_TAGGING
+        if (options->vlan_tagging && dev != DEV_TYPE_TAP)
+        {
+            msg (M_USAGE, "--vlan-tagging must be used with --dev tap");
+        }
+        if (!options->vlan_tagging)
+        {
+            if (options->vlan_accept != defaults.vlan_accept)
+            {
+                msg (M_USAGE, "--vlan-accept requires --vlan-tagging");
+            }
+            if (options->vlan_pvid != defaults.vlan_pvid)
+            {
+                msg (M_USAGE, "--vlan-pvid requires --vlan-tagging");
+            }
+        }
```

Comment:
don't we need some more checks here? for example: do all the `--vlan-accept` modes work fine with any `--vlan_pvid` value?

I should check this myself, but maybe you got an answer already: what happens if we specify `--vlan_pvid` in the client context but `--vlan-tagging` is not enabled in the global one? Maybe this checks apply to that too (as I Said, I'd need to check better)